File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0, maxW = 0, maxH = 0;
    
    for(int i=0; i<sizes.size(); i++){
        int w = sizes[i][0];
        int h = sizes[i][1];
        
        if(w < h) {
            int idx = w;
            w = h;
            h = idx;
        }
        
        if(maxW < w) maxW = w;
        if(maxH < h) maxH = h;
    }
    answer = maxW * maxH;
    
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {    
    let maxW = 0, maxH = 0;
    
    for(let i=0; i<sizes.length; i++){
        sizes[i].sort((a,b)=> a-b);
        
        if(sizes[i][0] > maxW) maxW = sizes[i][0];
        if(sizes[i][1] > maxH) maxH = sizes[i][1];
    }
    return maxW * maxH;
}